### PR TITLE
code cleanup

### PR DESCRIPTION
**Pull request by prieston**

### CODE REVIEW

File: src/compositeAttributes.js [hunk 1]
```diff
@@ -1,14 +1,14 @@
 module.exports = {
```
Comment (alexdor) [inline src/compositeAttributes.js:1]: module exports is a node thing, refactor this to use exports

File: src/Incidents/VideoPlay.js [hunk 2]
```diff
@@ -5,14 +5,13 @@ class VideoPlay extends MC.MediaPlayback {
     const video = this.element.entity.video;
     const ctx = this.element.entity.ctx;
 
-    // const playPromise = video.play();
     video.play();
     if (this.hasSetWaitingListener !== true) {
-      video.addEventListener("waiting", this._waitingHandler.bind(this));
+      video.addEventListener("waiting", this.waitingHandler.bind(this));
```
Comment (alexdor) [inline src/Incidents/VideoPlay.js:10]: shouldn't this event listener be stored so we can destroy it during cleanup?

File: src/Incidents/VideoPlay.js [hunk 3]
```diff
@@ -5,14 +5,13 @@ class VideoPlay extends MC.MediaPlayback {
...
       this.hasSetWaitingListener = true;
     }
     if (this.hasSetCanplayListener !== true) {
-      video.addEventListener("canplay", this._canplayHandler.bind(this));
+      video.addEventListener("canplay", this.canplayHandler.bind(this));
```
Comment (alexdor) [inline src/Incidents/VideoPlay.js:14]: shouldn't this event listener be stored so we can destroy it during cleanup?

File: src/Incidents/Effect.js [hunk 4]
```diff
@@ -4,51 +4,51 @@ const effects = require("../compositeAttributes").filter;
...
+      blur: 0,
       sepia: 0,
+      invert: 0,
+      grayscale: 0,
+      "hue-rotate": 0,
     };
   }
 
-  get _effectsUnits() {
+  get effectsUnits() {
```
Comment (alexdor) [inline src/Incidents/Effect.js:19]: Does this need to be a getter? Can't it just be a simple object?

**Review by alexdor — CHANGES_REQUESTED** [review]
Nicely done : ) Just some minor issues

Could you also extract the drawFrame function to a class function?

File: src/VideoClip.js [hunk 5]
```diff
@@ -2,49 +2,44 @@ const MC = require("@kissmybutton/motorcortex");
...
 
   onAfterRender() {
     const video = this.context.getElements("video")[0];
     video.muted = true;
     const canvas = this.context.getElements("canvas")[0];
     const ctx = canvas.getContext("2d");
-
+    const scaleX = this.width / video.videoWidth;
+    const scaleY = this.width / video.videoWidth;
     video.addEventListener(
```
Comment (alexdor) [inline src/VideoClip.js:39]: This event listener seems also to not being cleaned up at the end

**Review by alexdor — CHANGES_REQUESTED** [review]

File: src/VideoClip.js [hunk 6]
```diff
@@ -2,49 +2,44 @@ const MC = require("@kissmybutton/motorcortex");
...
+      <div>
+          <video id="video" style="${videoStyle}" preload="auto">
+              ${videoSources}
+          </video>
+          <canvas id="canvas"></canvas>
+      </div>
+    `;
   }
 
   get css() {
```
Comment (alexdor) [inline src/VideoClip.js:24]: can't we avoid the getter here and just use a static value?

**Review by alexdor — CHANGES_REQUESTED** [review]

File: src/VideoClip.js (hunk 6 above)
Comment (prieston) [reply src/VideoClip.js:24]: I think not. We should ask trad

Comment (alexdor) [reply src/VideoClip.js:24]: For the consumer of this (MC in this case) there shouldn't be any difference between a getter that has a static value and just assigning that value. Meaning the following two examples should be the same
```js
class Smth{
  get css(){ return 'css1'}
}
```

```js
class Smth{
  css = 'css1'
}
```

@andreas-trad any thoughts?

**Pull request merged**